Make UCP option the default for regex matching

Keno opened the pull request:
Fixes #27084. Regexes now match based on unicode character properties,
rather than just ASCII character properties, e.g. `match(r"\w+", "café")`
will now match the entire word (and not just `caf`). This behavior can
be disabled with the `a` flag to the regex string macro (e.g. `r"\w+"a`).

Keno committed:
Make UCP option the default for regex matching

Fixes #27084. Regexes now match based on unicode character properties,
rather than just ASCII character properties, e.g. `match(r"\w+", "café")`
will now match the entire word (and not just `caf`). This behavior can
be disabled with the `a` flag to the regex string macro (e.g. `r"\w+"a`).

nalimilan commented on base/regex.jl:
```diff
@@ -72,8 +76,12 @@ after the ending quote, to change its behaviour:
 - `x` enables "comment mode": whitespace is enabled except when escaped with `\\`, and `#`
   is treated as starting a comment.
+- `a` disables `UCP` mode (enables ASCII mode). By default `\\B`, `\\b`, `\\D`, `\\d`, `\\S`,
+  `\\s`, `\\W`, `\\w`, etc match based on unicode character properties. With this option,
```
"etc." and "Unicode". "are recognized" -> "these sequences only match"

Keno committed:
Fix Windows path regex

\w matches numbers as well (which are not valid drive letters) and will match unicode
characters once we turn on UCP.
